packages/storybook/stories/tabs.stories.tsx

```diff
@@ -172,25 +172,19 @@ export const UnloadTabCallback: Story = () => {
             title: 'First Button',
             panelContent: <StyledDiv>First tab content</StyledDiv>,
             onBeforeUnload: () => {
-                console.info('cannot change tab because callback return false');
-                return false;
+                console.info('cannot change tab because onBeforeUnload promise resolves to false here');
+                return new Promise<boolean>((resolve) => resolve(false));
             },
         },
         {
             title: 'Second Button',
             panelContent: <StyledDiv>Second tab content</StyledDiv>,
-            onBeforeUnload: () => {
-                console.info('second tab unload confirmed');
-                return true;
-            },
+            onBeforeUnload: () => new Promise<boolean>((resolve) => resolve(true)),
         },
         {
             title: 'Third Button',
             panelContent: <StyledDiv>Third tab content</StyledDiv>,
-            onBeforeUnload: () => {
-                console.info('third tab unload confirmed');
-                return true;
-            },
+            onBeforeUnload: () => new Promise<boolean>((resolve) => resolve(true)),
```

Comment:
Pas certain de cette story là. Ça donne l'impression que le component est brisé, on peut pas changer de Tab.

Reply:
j'ai mis un titre qui parle sur la tab : 'Tab that cannot change because onBeforeUnload resolves to false'

Reply:
<img width="1079" alt="Capture d’écran, le 2022-04-01 à 10 08 28" src="https://user-images.githubusercontent.com/19400226/161280222-80523689-fff2-49c5-a013-bb55962cd102.png">